### ISSUE TITLE
Closing WebSocket prior to Ready event

### ISSUE THREAD

**Issue by gmyers18** [wildfoundry/dataplicity-lomond#60]
I think I've run into a bug or two with 0.2.4 when calling `WebSocket.close()` prior to a `Ready` event.  Consider the following example code

```python
import logging

from lomond import WebSocket, events

logging.basicConfig(format='%(message)s', level=logging.DEBUG)
ws = WebSocket('ws://echo.websocket.org')

for event in ws:
    logging.info(event)
    if isinstance(event, events.Connecting):
        logging.info('Closing @Connecting...')
        ws.close()
    if isinstance(event, events.Connected):
        logging.info('Closing @Connected...')
        ws.close()
```

which produces
```
Connecting(url='ws://echo.websocket.org')
Closing @Connecting...
WebSocket unavailable; data not sent
Traceback (most recent call last):
  File "test.py", line 12, in <module>
    ws.close()
  File "lomond\websocket.py", line 234, in close
    self.state.sent_close_time = self.session.session_time
  File "lomond\session.py", line 60, in session_time
    return time.time() - self._start_time
TypeError: unsupported operand type(s) for -: 'float' and 'NoneType'
```

The exception from the Connecting stage can be remedied by checking the return value from `_send_close()` when called in `WebSocket.close()` and conditionally not setting the state when the return value is `False`.  This will get you further along to the Connected stage, where the same exception will occur.  The solution here is less clear.  The root problem is due to session time not being available until a `Ready` event is handled at which point `WebSocketSession._start_time` is set.

My use case has the event loop running in a thread and I'd like to be able to exit the event loop gracefully (and subsequently exit the thread) by calling `WebSocket.close()` from the main thread.  But, in some corner cases I try to tear things down during either the Connecting or Connected stage.  Even assuming a fix to the issues above, there is no actual connection during the Connecting stage so there is nothing to close.  Is there a recommendation for a way to gracefully exit the event loop in a situation like this?  My only idea right now if to repetitively call `close()`in a loop until `WebSocket.sent_close_time` is not `None`.

**Comment by willmcgugan** [comment]
Please give 0.2.5a0 a try. It should handle both the connecting and connected cases gracefully.

https://github.com/wildfoundry/dataplicity-lomond/pull/61

**Comment by gmyers18** [comment]
@willmcgugan Thanks for the quick fix!  I tested with the new release and I think it resolves all of my issues/questions.

**Comment by willmcgugan** [comment]
Fixed in 0.2.5
